src/Expression.php

```diff
@@ -13,6 +13,11 @@
  */
 class Expression implements \ArrayAccess, \IteratorAggregate
 {
+    public const ESCAPE_PARAM = 'param';
```

Comment:
This is internal only, can it be protected?

Also, annotate like:
`/** @const string "[]" in template, escape as string */`
`/** @const string "[]" in template, escape as identifier */`

Reply:
I think it should be more like STRING, IDENTIFIER, IDENTIFIER_SOFT, NONE